### PR TITLE
[WIP] Switch bytecounts page to websockets

### PR DESCRIPTION
**Pull request by ThePletch**
Included in this is the front-end Javascript and the beginnings of the configuration for mod_websocket - not currently completed, but mod_websocket installation is in progress.

### CODE REVIEW

File: app/js/request.js [hunk 1]
```diff
@@ -26,8 +26,37 @@ var requestModule = (function(){
...
+    }, keepAliveInterval);
+  };
+
+  var keepAlive = function(socket, interval){
+    sendSocketMessage(socket);
+  };
+
+  var sendSocketMessage = function(socket, message){
+    var parcel = {
+      csrf: getCsrfToken(),
```
Comment (ThePletch) [inline app/js/request.js]: This should ultimately be unnecessary when the websocket server is fully configured.

**Comment by ThePletch** [comment]
I'm deeply apologetic about this, but I think I lack the server expertise to actually finish this - I've done as much configuration as I can without actually installing/testing `mod_websocket`, but I can't find any information online about its functionality and I've hit a dead end with installing it after sinking about ten hours into it.

I hope this code is of some use to a future developer.

**Comment by Rangak** [comment]
Not a problem. Thanks for bringing it this far. I think you are most of the way there so I will finish it from here on. This is an important improvement and your contribution is much appreciated.

File: lighttpd/lighttpd.conf.template [hunk 2]
```diff
@@ -93,5 +94,18 @@ server.groupname = "%(GROUP)s"
...
+                    (
+                     "host" => "127.0.0.1",
+                     "port" => "80"
+                    )
+                   )
+
+# @ThePletch : Timeout for websocket connections in seconds
+websocket.timeout = 20
+
+websocket.debug = 4
```
Comment (ThePletch) [inline lighttpd/lighttpd.conf.template]: values:
0 - disable logging
1 - log fatal messages
2 - log error messages
3 - log warning messages
4 - log debug messages

each logs anything that falls under a log level less than the current setting

**Pull request closed without merging**